src/pkg/packager/lint/validator.go

```diff
@@ -7,62 +7,175 @@ package lint
+
+const (
+	validationError   ValidationType = 1
+	validationWarning ValidationType = 2
+)
+
+type packageKey struct {
+	path string
+	name string
+}
+
+type validatorMessage struct {
+	yqPath         string
+	description    string
+	item           string
+	packageKey     packageKey
+	validationType ValidationType
+}
+
+func (vt ValidationType) String() string {
+	if vt == validationError {
+		return utils.ColorWrap("Error", color.FgRed)
+	} else if vt == validationWarning {
+		return utils.ColorWrap("Warning", color.FgYellow)
+	} else {
+		return ""
+	}
+}
+
+func (v validatorMessage) String() string {
+	if v.item != "" {
+		v.item = fmt.Sprintf(" - %s", v.item)
+	}
+	if v.packageKey.path == "" && v.yqPath == "" && v.item == "" {
+		return v.description
+	}
+	return fmt.Sprintf("%s%s", v.description, v.item)
+}
+
 // Validator holds the warnings/errors and messaging that we get from validation
 type Validator struct {
-	warnings           []string
-	errors             []error
+	findings           []validatorMessage
 	jsonSchema         []byte
 	typedZarfPackage   types.ZarfPackage
 	untypedZarfPackage interface{}
 }
 
 // DisplayFormattedMessage message sent to user based on validator results
 func (v Validator) DisplayFormattedMessage() {
-	if !v.hasWarnings() && !v.hasErrors() {
-		message.Successf("Schema validation successful for %q", v.typedZarfPackage.Metadata.Name)
+	if !v.hasFindings() {
+		message.Successf("0 findings for %q", v.typedZarfPackage.Metadata.Name)
 	}
 	v.printValidationTable()
 }
 
 // IsSuccess returns true if there are not any errors
 func (v Validator) IsSuccess() bool {
-	return !v.hasErrors()
+	for _, finding := range v.findings {
+		if finding.validationType == validationError {
+			return false
+		}
+	}
+	return true
+}
+
+func (v Validator) HasUnsetVarMessageForPkg(pk packageKey) bool {
+	for _, finding := range v.findings {
+		if finding.description == lang.UnsetVarLintWarning && finding.packageKey == pk {
+			return true
+		}
+	}
+	return false
 }
 
 func (v Validator) printValidationTable() {
-	if v.hasWarnings() || v.hasErrors() {
-		header := []string{"Type", "Message"}
-		connectData := [][]string{}
-		for _, warning := range v.warnings {
-			connectData = append(connectData, []string{utils.ColorWrap("Warning", color.FgYellow), warning})
+	if !v.hasFindings() {
+		return
+	}
+
+	packageKeys := v.getUniquePackageKeys()
+	connectData := make(map[packageKey][][]string)
+	for _, packageKey := range packageKeys {
+		header := []string{"Type", "Path", "Message"}
+
+		for _, finding := range v.findings {
+			if finding.packageKey == packageKey {
+				connectData[packageKey] = append(connectData[packageKey],
+					[]string{finding.validationType.String(), finding.getPath(), finding.String()})
+			}
 		}
-		for _, err := range v.errors {
-			connectData = append(connectData, []string{utils.ColorWrap("Error", color.FgRed), err.Error()})
+		//We should probably move this println into info
+		pterm.Println()
```

Comment:
Thoughts on adding `pterm.Println()` to `message.info()` @Racer159 ? I know you mentioned you wanted to add new lines on messages generally, not sure if we hadn't gotten around to adding it to info or are not adding it for a reason

Reply:
Could we use note?